### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by ranocha**
I enabled GitHub actions for CI and Documenter. It would be good if someone with more access rights can check this and set the appropriate secrets/settings.

I want to add this to allow more reliable testing, e.g. to bump the compat bound of BenchmarkTools.jl (#124).

### CODE REVIEW

**Comment by ranocha** [comment]
Bump. Is there any interest in this?

**Comment by KristofferC** [comment]
Yes, thanks. I wonder, is there a way not to get 7 coveralls checks? It feels a bit excessive.

**Comment by ranocha** [comment]
> I wonder, is there a way not to get 7 coveralls checks? It feels a bit excessive.

I don't know... I would really like to get rid of the intermediate runs and only keep `coverage/coveralls`, but I couldn't find a way to achieve that...

**Comment by KristofferC** [comment]
Maybe we can switch to codecov then? That seems to only give one check. I think this: https://github.com/KristofferC/NearestNeighbors.jl/blob/2efd998b6bb74d175e6cbbe1c8bc8e775e502f0b/.github/workflows/CI.yml#L45-L48 is the relevant lines that need to be used.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          restore-keys: |
+            ${{ runner.os }}-test-${{ env.cache-name }}-
+            ${{ runner.os }}-test-
+            ${{ runner.os }}-
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
+      - uses: julia-actions/julia-processcoverage@v1
+      - uses: codecov/codecov-action@v1
+        with:
+          file: lcov.info
```
Comment (ranocha) [inline .github/workflows/CI.yml:50]: @KristofferC These lines enable Codecov - I guess somebody needs to enable Codecov on their website?

**Comment by ranocha** [comment]
> Maybe we can switch to codecov then? That seems to only give one check. I think this: https://github.com/KristofferC/NearestNeighbors.jl/blob/2efd998b6bb74d175e6cbbe1c8bc8e775e502f0b/.github/workflows/CI.yml#L45-L48 is the relevant lines that need to be used.

We observed that Codecov and Coveralls report different coverage percentages when collecting data from parallel Julia tests, see https://community.codecov.io/t/code-lines-wrongly-detected-as-executable-in-parallel-jobs-for-a-julia-project/1940. Thus, I think it might be worth keeping the additional information reported by Coveralls

**Comment by KristofferC** [comment]
These tests are not run in parallel so I think that is no worry.

I think codecov is already active because it has data for the latest commit: https://codecov.io/gh/JuliaCI/PkgBenchmark.jl/commit/21896fc9a4ff5d0d26db086aa7ed72a9f88b1cb9/

**Comment by ranocha** [comment]
Is there anything left to do for me?
- I would propose to keep Coveralls since additional information doesn't hurt.
- Someone could enable the Codecov integration in PRs if desired (report of the bot).
- Someone should disable Appveyor to remove the failing CI test.

**Comment by KristofferC** [comment]
I've removed AV and coveralls. Additional information does hurt if is too noisy.

Thanks for the help with this!

**Pull request merged**